src/librustc/traits/project.rs

```diff
@@ -1231,6 +1230,8 @@ fn confirm_select_candidate<'cx, 'gcx, 'tcx>(
             confirm_closure_candidate(selcx, obligation, data),
         super::VtableFnPointer(data) =>
             confirm_fn_pointer_candidate(selcx, obligation, data),
+        super::VtableTraitAlias(data) =>
+            confirm_trait_alias_candidate(selcx, obligation, data),
         super::VtableObject(_) =>
             confirm_object_candidate(selcx, obligation, obligation_trait_ref),
         super::VtableAutoImpl(..) |
```

Comment:
As we said on Zulip, trait aliases don't define associated types, so I think you can lump `VtableTraitAlias` in with these other cases.